### ISSUE TITLE
panic: version string "v20220409-v1.23.0" doesn't match expected regular expression

### ISSUE THREAD

**Issue by hakman** [kubernetes/cloud-provider-aws#339]
**What happened**:
Tests fail because `aws-cloud-controller-manager` fail to start:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kops/13452/pull-kops-e2e-cni-cilium-ipv6/1516070708126420992
```
==== START logs for container aws-cloud-controller-manager of pod kube-system/aws-cloud-controller-manager-94ngm ====
panic: version string "v20220409-v1.23.0" doesn't match expected regular expression: "^v(\d+\.\d+\.\d+)"

goroutine 1 [running]:
k8s.io/component-base/metrics.parseVersion({{0x0, 0x0}, {0x0, 0x0}, {0x29703c0, 0x11}, {0x254765f, 0xb}, {0x0, 0x0}, ...})
	k8s.io/component-base@v0.23.5/metrics/version_parser.go:47 +0x27e
k8s.io/component-base/metrics.newKubeRegistry({{0x0, 0x0}, {0x0, 0x0}, {0x29703c0, 0x11}, {0x254765f, 0xb}, {0x0, 0x0}, ...})
	k8s.io/component-base@v0.23.5/metrics/registry.go:320 +0x118
k8s.io/component-base/metrics.NewKubeRegistry()
	k8s.io/component-base@v0.23.5/metrics/registry.go:335 +0x78
k8s.io/component-base/metrics/legacyregistry.init()
	k8s.io/component-base@v0.23.5/metrics/legacyregistry/registry.go:29 +0x1d
==== END logs for container aws-cloud-controller-manager of pod kube-system/aws-cloud-controller-manager-94ngm ====
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:
Try to start `aws-cloud-controller-manager` using the latest stable image: `registry.k8s.io/provider-aws/cloud-controller-manager:v1.23.0`.

**Anything else we need to know?**:
Guess it's related to: https://github.com/kubernetes/cloud-provider-aws/pull/317.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.23.5
- Cloud provider or hardware configuration: AWS
- OS (e.g. from /etc/os-release): Ubuntu 22.04
- Kernel (e.g. `uname -a`):
- Install tools: kOps
- Others: n/a

/kind bug

**Comment by olemarkus** [comment]
Also related to https://github.com/kubernetes/cloud-provider-aws/pull/324

**Comment by rifelpet** [comment]
I tracked down the source of the version string to this:

https://github.com/kubernetes/test-infra/blob/0ed976e5a2f3007e055a84b3900b806bd17b57ee/config/jobs/image-pushing/README.md#custom-substitutions

https://github.com/kubernetes/test-infra/blob/0ed976e5a2f3007e055a84b3900b806bd17b57ee/images/builder/main.go#L95-L108

**Comment by hakman** [comment]
Unfortunately this is not yet fixed by #350:
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/cloud-provider-aws-push-images/1523933803632922624
```
Error: failed to solve: process "/bin/sh -c GO111MODULE=on CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} GOPROXY=${GOPROXY} 
\t\tgo build \t\t-trimpath \t\t-ldflags=\"-w -s -X k8s.io/component-base/version.gitVersion=${VERSION}\"
\t\t-o=aws-cloud-controller-manager \t\tcmd/aws-cloud-controller-manager/main.go"
did not complete successfully: exit code: 2
Usage:
  /root/.docker/cli-plugins/docker-buildx build [OPTIONS] PATH | URL | - [flags]
```
